File: 3517. Smallest Palindromic Rearrangement I.cpp
class Solution {
public:
    string smallestPalindrome(string s) 
    {
        vector<int> count(26);
        for(auto &ch : s) count[ch - 'a'] += 1;

        int cur = 0, n = s.size();
        for(int i = 0; i < 26; i++)
        {
            if(count[i] % 2) s[n / 2] = i + 'a';
            while(count[i] > 1)
            {
                s[cur] = i + 'a';
                s[n - cur - 1] = i + 'a';
                count[i] -= 2;
                cur += 1;
            }
        }   
        return s; 
    }
};

File: 3517. Smallest Palindromic Rearrangement I.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        n = len(s)
        ans = [''] * n

        i = 0
        store = Counter(s)
        
        for ch in ascii_lowercase:
            if store[ch] % 2:
                ans[n // 2] = ch
            
            while store[ch] > 1:
                ans[i] = ans[n - i - 1] = ch
                store[ch] -= 2
                i += 1
                
        return ''.join(ans)